### DISCUSSION TITLE
Documentation for Parsing Dotnet Resources

### DISCUSSION THREAD

**Discussion by utkonos** [icsharpcode/ILSpy#3055] [Help] [answered]
Can someone point to documentation for the structure of a dotnet resource: header and data?

I am able to save a resource using ILSpy here:

<img width="1061" alt="image" src="https://github.com/icsharpcode/ILSpy/assets/2775739/a339f53b-51fd-4b51-a84c-aae398e6d654">

But then I would like to be able to parse the resulting file. There is clearly a header and then PNG data in this example, but I would like to find the documentation for this format in general. Any pointers would be appreciated.

<img width="561" alt="image" src="https://github.com/icsharpcode/ILSpy/assets/2775739/40a74b12-cd13-430f-9ae5-475870d6eb88">

**Comment by dgrunwald** [comment, marked as answer]
I'm not sure if any documentation exists.
The class System.Resources.ResourceReader implements the logic to read these files. ILSpy uses its own [modified copy](https://github.com/icsharpcode/ILSpy/blob/master/ICSharpCode.Decompiler/Util/ResourcesFile.cs) of this logic to avoid the security issues with deserialization.

**Comment by utkonos** [comment]
@dgrunwald thanks for the quick answer. I now have a few sources of information and some implementations.

This is the class you're referring to:
https://learn.microsoft.com/en-us/dotnet/api/system.resources.resourcereader

These are two source code files I can look at as well as the implementation in ILSpy:
https://github.com/microsoft/referencesource/blob/master/mscorlib/system/resources/resourcewriter.cs
https://github.com/microsoft/referencesource/blob/master/mscorlib/system/resources/resourcereader.cs

I will also dig into the source code for Mono and see if there is an implementation of the same there.
